#! /bin/sh -e

# All lines beginning with `# DP:' are a description of the patch.
# DP: Description: linuxthreads/nptl pthread_atfork.c in libpthread_nonshared.a 
# DP:		   is compiled with -fPIC, not -fpic to fix lam build error.
# DP: Related bugs: #280445: libpthread miscompiled (wrong pic options) on s390
# DP: Dpatch author: GOTO Masanori <gotom@debian.org>
# DP: Patch author: GOTO Masanori <gotom@debian.org>
# DP: Upstream status: Not submitted
# DP: Status Details: 
# DP: Date: 2004-12-02

PATCHLEVEL=0

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p$PATCHLEVEL < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p$PATCHLEVEL < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- linuxthreads/sysdeps/s390/Makefile.org	2004-11-29 11:33:03.000000000 -0500
+++ linuxthreads/sysdeps/s390/Makefile	2004-11-29 11:31:27.000000000 -0500
@@ -3,4 +3,9 @@
 endif
 ifeq ($(subdir),linuxthreads)
 libpthread-routines += ptw-sysdep
+object-suffixes-left := $(libpthread-nonshared)
+define o-iterator-doit
+$(objpfx)$o.os: pic-ccflag = -fPIC
+endef
+include $(o-iterator)
 endif
--- nptl/sysdeps/s390/Makefile.org	2004-11-30 18:45:18.000000000 -0500
+++ nptl/sysdeps/s390/Makefile	2004-11-30 18:45:34.000000000 -0500
@@ -22,4 +22,9 @@
 
 ifeq ($(subdir),nptl)
 libpthread-routines += ptw-sysdep
+object-suffixes-left := $(libpthread-nonshared)
+define o-iterator-doit
+$(objpfx)$o.os: pic-ccflag = -fPIC
+endef
+include $(o-iterator)
 endif
